### PR TITLE
Fixing 5 entities according to new search approach

### PR DESCRIPTION
**Pull request by oshtaier**
**Subnet:**
```
nosetests tests/foreman/ui/test_subnet.py
....S.....
----------------------------------------------------------------------
Ran 10 tests in 405.892s

OK (SKIP=1)
```
**Location:**
```
nosetests tests/foreman/ui/test_location.py
.E..................E......
----------------------------------------------------------------------
Ran 27 tests in 4338.874s

FAILED (errors=2)
```
```
nosetests tests/foreman/ui/test_location.py -m test_add_configtemplate
.
----------------------------------------------------------------------
Ran 1 test in 239.159s

OK
```
```
nosetests tests/foreman/ui/test_location.py -m test_remove_configtemplate
.
----------------------------------------------------------------------
Ran 1 test in 365.310s

OK
```

**Lifecycle Environment:**
```
nosetests tests/foreman/ui/test_contentenv.py
....
----------------------------------------------------------------------
Ran 4 tests in 213.725s

OK
```

**LDAP:**
```
nosetests tests/foreman/ui/test_ldapauthsource.py -m test_delete_ldap_auth_withad
.
----------------------------------------------------------------------
Ran 1 test in 213.554s

OK
```
```
nosetests tests/foreman/ui/test_ldapauthsource.py -m test_create_ldap_authsource_withad
.
----------------------------------------------------------------------
Ran 1 test in 81.151s

OK
```

### CODE REVIEW

File: robottelo.properties.sample [hunk 1]
```diff
@@ -68,7 +68,7 @@ ssh_key=
 # [ldap]
 # hostname=
 # username=
-# passwd=
+# password=
```
Comment (abalakh) [inline robottelo.properties.sample]: :+1:

File: robottelo/ui/ldapauthsource.py [hunk 2]
```diff
@@ -52,25 +60,14 @@ def create(self, name=None, server=None, ldaps=False, port=None,
...
-                u'Could not search the entity "{0}".'
-                .format(name)
-            )
-        return element
+        self.navigate_to_entity()
+        strategy1, value1 = self._search_locator()
+        return self.wait_until_element((strategy1, value1 % (name, name)))
 
-    def delete(self, name, really=False):
+    def delete(self, name, really=True):
```
Comment (abalakh) [inline robottelo/ui/ldapauthsource.py]: :+1:

File: robottelo/ui/subnet.py [hunk 3]
```diff
@@ -60,19 +70,13 @@ def delete(self, subnet_name, really=True):
         self.delete_entity(
             subnet_name,
             really,
-            locators['subnet.display_name'],
             locators['subnet.delete'],
         )
 
-    def search_subnet(self, subnet_name):
+    def search_subnet_and_validate(self, subnet_name):
```
Comment (abalakh) [inline robottelo/ui/subnet.py]: Pretty long name. As it's already a subnet module, maybe rename it to `search_and_validate`? Not a blocker

File: robottelo/ui/systemgroup.py [hunk 4]
```diff
@@ -4,9 +4,19 @@
...
 class SystemGroup(Base):
+    is_katello = True
+
+    def navigate_to_entity(self):
+        """Navigate to Host Collection entity page"""
+        Navigator.go_to_host_collections()
+
+    def _search_locator(self):
+        """Specify locator for Host Collection entity search procedure"""
+        return locators['system-groups.search']
```
Comment (abalakh) [inline robottelo/ui/systemgroup.py]: Surprised this locator is called `system-groups.search`, not `system-groups.select` or smth. Double checked, this is a right one :+1:

File: tests/foreman/ui/test_contentenv.py [hunk 5]
```diff
@@ -17,6 +16,7 @@ class ContentEnvironment(UITestCase):
 
     @classmethod
     def setUpClass(cls):
+        super(ContentEnvironment, cls).setUpClass()
```
Comment (abalakh) [inline tests/foreman/ui/test_contentenv.py]: :+1:

File: tests/foreman/ui/test_subnet.py [hunk 6]
```diff
@@ -49,9 +49,7 @@ def test_create_subnet_with_different_names(self):
                         subnet_network=gen_ipaddr(ip3=True),
                         subnet_mask=gen_netmask(),
                     )
-                    self.assertIsNotNone(
-                        self.subnet.search_subnet(subnet_name=name)
-                    )
+                    self.assertIsNotNone(self.subnet.search(name))
```
Comment (abalakh) [inline tests/foreman/ui/test_subnet.py]: Here and till the end of module there was originally `search_and_validate`, not just regular `search` func. Was that changed on purpose?

Comment (oshtaier) [inline tests/foreman/ui/test_subnet.py]: Of course. We need that special method only when we need to get specific attribute for validation

**Comment by abalakh** [comment]
ACK. 
Very nice PR, with few important bugfixes :+1:

File: robottelo.properties.sample (hunk 1 above)
Comment (elyezer) [inline robottelo.properties.sample]: :+1:

File: robottelo/ui/contentenv.py [hunk 7]
```diff
@@ -24,16 +33,28 @@ def create(self, name, description=None, prior=None):
                 common_locators['description'], description)
         self.click(common_locators['create'])
 
+    def search(self, name):
```
Comment (elyezer) [inline robottelo/ui/contentenv.py]: No super call?

Comment (elyezer) [inline robottelo/ui/contentenv.py]: Also, this overridden search could not be dropped in favor of the base one?

File: robottelo/ui/ldapauthsource.py [hunk 8]
```diff
@@ -52,25 +60,14 @@ def create(self, name=None, server=None, ldaps=False, port=None,
 
     def search(self, name):
```
Comment (elyezer) [inline robottelo/ui/ldapauthsource.py]: This search can be dropped in favor of the base one?

File: tests/foreman/ui/test_contentenv.py (hunk 5 above)
Comment (elyezer) [inline tests/foreman/ui/test_contentenv.py]: :+1:

**Comment by elyezer** [comment]
ACK, pending comments/questions.

File: robottelo/ui/contentenv.py (hunk 7 above)
Comment (abalakh) [inline robottelo/ui/contentenv.py]: The page doesn't contain search bar, all this method is doing is just navigating the page and selecting the right link. That's why no super call and base search (which is designed to work with search bar) can't be used.

File: robottelo/ui/ldapauthsource.py (hunk 8 above)
Comment (abalakh) [inline robottelo/ui/ldapauthsource.py]: The page doesn't contain search bar, all this method is doing is just navigating the page and selecting the right link. That's why no super call and base search (which is designed to work with search bar) can't be used.

Comment (oshtaier) [inline robottelo/ui/ldapauthsource.py]: Nope, it is a corner case without search box or search button

File: robottelo/ui/contentenv.py (hunk 7 above)
Comment (oshtaier) [inline robottelo/ui/contentenv.py]: Nope, it is a corner case without search box or search button. Or even entities list :)

**Comment by oshtaier** [comment]
@elyezer let me know if you have any comments/questions

**Comment by kbidarkar** [comment]
We have good bug fixes and refactoring for the search. The changes look good.
ACK

**Comment by kbidarkar** [comment]
The comments and questions have been answered and are the changes are fine as:
a) no searchbox
b) and hence, no call to super 

merging it.

File: robottelo/ui/ldapauthsource.py (hunk 8 above)
Comment (elyezer) [inline robottelo/ui/ldapauthsource.py]: Ok, I would ask for make this clear in form of comments/docstring. But this PR is already merged.

If you want to add this later so we can remember why it is needed.

Comment (oshtaier) [inline robottelo/ui/ldapauthsource.py]: sure @elyezer. I will take that into account before merge into master

Comment (elyezer) [inline robottelo/ui/ldapauthsource.py]: @oshtaier thank you very much!

**Pull request merged**